src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/IHeartbeatProvider.cs

```diff
@@ -0,0 +1,19 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+    internal interface IHeartbeatProvider
+    {
+        string DiagnosticsInstrumentationKey { set; }
+
+        bool AddHealthProperty(string name, string value, bool isHealthy);
```

Comment:
think i mentioned this somewhere else, but its easier if the parameters are named like healthPropertyName, healthPropertyValue etc, in all methods in this interface/impl